app/javascript/turbo/cable_stream_source_element.js

```diff
@@ -5,7 +5,11 @@ import snakeize from "./snakeize"
 class TurboCableStreamSourceElement extends HTMLElement {
   async connectedCallback() {
     connectStreamSource(this)
-    this.subscription = await subscribeTo(this.channel, { received: this.dispatchMessageEvent.bind(this) })
+    this.subscription = await subscribeTo(this.channel, {
+      received: this.dispatchMessageEvent.bind(this),
+      connected: this.subscriptionConnected.bind(this),
+      disconnected: this.subscriptionDisconnected.bind(this),
```

Comment:
Not for this PR, but we might be able to use fat arrows instead of `bind(this)`. Caveat is that [public class fields are only available in Safari > 14.1](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields).

Reply:
BTW, no extra comma at the end of the object literal.

Reply:
I don't think these are "public class fields", they are just regular methods on the class

Reply:
@marcoroth I mean that you can direclt define an event handler bound to this within a class using a fat arrow:

```js
class EventHandler {
  doSomething = (event) => {
    ...
  }
}
```

But that's implicitly using the public class fields feature.

Reply:
@afcapel ah, I see what you mean, I misunderstood! Yeah, that's right 👍🏼